### PR TITLE
Clone bugfix/st synthetic named expr in match

### PR DESCRIPTION
**Pull request by visz11**
## **CodeAnt-AI Description**
Fix match subject type inference for complex expressions

### What Changed
- When a match's subject is a complex expression (calls, attributes, arithmetic, dict/item access, unary ops, await, etc.), the checker now creates a synthetic, unique named subject so pattern capture variables are inferred correctly and exhaustivity/unreachability checks are accurate.
- Primitive literals and already-named expressions are left as-is; inline collection literals are now allowed for inference too.
- Inferred type mappings for the synthetic subject are propagated back to the original subject expression when available.
- Tests updated: one previously xfailed test is enabled and multiple new match-case tests were added to cover function calls, attribute access, operations, dict item access, and literal edge cases.

### Impact
`✅ Clearer pattern capture types in match statements`
`✅ Fewer false "unreachable" match branches`
`✅ More accurate reveal_type results in match tests`
<details>
<summary><strong>💡 Usage Guide</strong></summary>

### Checking Your Pull Request
Every time you make a pull request, our system automatically looks through it. We check for security issues, mistakes in how you're setting up your infrastructure, and common code problems. We do this to make sure your changes are solid and won't cause any trouble later.

### Talking to CodeAnt AI
Got a question or need a hand with something in your pull request? You can easily get in touch with CodeAnt AI right here. Just type the following in a comment on your pull request, and replace "Your question here" with whatever you want to ask:
<pre>
<code>@codeant-ai ask: Your question here</code>
</pre>
This lets you have a chat with CodeAnt AI about your pull request, making it easier to understand and improve your code.

#### Example
<pre>
<code>@codeant-ai ask: Can you suggest a safer alternative to storing this secret?</code>
</pre>

### Preserve Org Learnings with CodeAnt
You can record team preferences so CodeAnt AI applies them in future reviews. Reply directly to the specific CodeAnt AI suggestion (in the same thread) and replace "Your feedback here" with your input:
<pre>
<code>@codeant-ai: Your feedback here</code>
</pre>
This helps CodeAnt AI learn and adapt to your team's coding style and standards.

#### Example
<pre>
<code>@codeant-ai: Do not flag unused imports.</code>
</pre>

### Retrigger review
Ask CodeAnt AI to review the PR again, by typing:
<pre>
<code>@codeant-ai: review</code>
</pre>

### Check Your Repository Health
To analyze the health of your code repository, visit our dashboard at [https://app.codeant.ai](https://app.codeant.ai). This tool helps you identify potential issues and areas for improvement in your codebase, ensuring your repository maintains high standards of code health.

</details>

## Summary by CodeRabbit

* **New Features**
  * Enhanced type checking for Python 3.10 match statements with improved handling of complex patterns including function calls, attributes, operations, and dictionary items.

* **Tests**
  * Expanded test coverage for pattern matching scenarios, including new cases for named tuples, literals, and various match statement constructs.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: mypy/checker.py [hunk 1]
```diff
@@ -67,6 +67,7 @@
     CallExpr,
     ClassDef,
     ComparisonExpr,
+    ComplexExpr,
```
[the remark this answers is not in the document]
Comment (visz11) [reply mypy/checker.py:70]: @refacto-chat you are giving same suggestion here , provide me correct fix?

[the remark this answers is not in the document]
Comment (visz11) [reply mypy/checker.py:70]: @refacto-chat you are giving same suggestion here , provide me correct fix?

**Comment by visz11** [comment]
@refacto-visz

File: mypy/checker.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (visz11) [reply mypy/checker.py:70]: @refacto-chat can you provide me fix code suggestion here

File: mypy/checker.py [hunk 2]
```diff
@@ -7885,6 +7912,12 @@ def warn_deprecated_overload_item(
                     if candidate == target:
                         self.warn_deprecated(item.func, context)
 
+    def new_unique_dummy_name(self, namespace: str) -> str:
+        """Generate a name that is guaranteed to be unique for this TypeChecker instance."""
+        name = f"dummy-{namespace}-{self._unique_id}"
```
[the remark this answers is not in the document]
Comment (visz11) [reply mypy/checker.py:7917]: @refacto-chat provide me commitable fix code here
